src/libfetchers/github.cc

```diff
@@ -145,7 +145,7 @@ struct GitArchiveInputScheme : InputScheme
         if (ref)
             path.push_back(*ref);
         if (rev)
-            path.push_back(rev->to_string(HashFormat::Base16, false));
+            path.push_back(abbreviate ? rev->gitShortRev() : rev->gitRev());
```

Comment:
minor observation: `gitShortRev()` hardcodes a length of 7.

I imagine this is because libgit doesn't give us a way to ask for "the shortest unique rev", but in case it's not, it would be great to fix that at some point...